src/transaction/input.rs

```diff
@@ -4,67 +4,27 @@
 // This SAFE Network Software is licensed under the BSD-3-Clause license.
 // Please see the LICENSE file for more details.
 
+use super::{Amount, Error, Result};
+use crate::{DbcId, DbcTransaction, DerivedKey};
 use blsttc::Signature;
-use bulletproofs::PedersenGens;
-
 #[cfg(feature = "serde")]
 use serde::{self, Deserialize, Serialize};
 
-use super::{Error, Result, RevealedAmount};
-use crate::{BlindedAmount, DbcId, DerivedKey};
-
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[derive(Debug, Clone)]
-pub struct RevealedInput {
-    #[cfg_attr(feature = "serde", serde(skip_serializing))]
-    pub derived_key: DerivedKey,
-    pub revealed_amount: RevealedAmount,
-}
-
-impl RevealedInput {
-    pub fn new(derived_key: DerivedKey, revealed_amount: RevealedAmount) -> Self {
-        Self {
-            derived_key,
-            revealed_amount,
-        }
-    }
-
-    pub fn dbc_id(&self) -> DbcId {
-        self.derived_key.dbc_id()
-    }
-
-    pub fn revealed_amount(&self) -> &RevealedAmount {
-        &self.revealed_amount
-    }
-
-    pub fn blinded_amount(&self, pc_gens: &PedersenGens) -> BlindedAmount {
-        self.revealed_amount.blinded_amount(pc_gens)
-    }
-
-    pub fn sign(&self, msg: &[u8], pc_gens: &PedersenGens) -> BlindedInput {
-        BlindedInput {
-            dbc_id: self.dbc_id(),
-            blinded_amount: self.blinded_amount(pc_gens),
-            signature: self.derived_key.sign(msg),
-        }
-    }
-}
-
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Eq, PartialEq, Debug, Clone)]
-pub struct BlindedInput {
+pub struct Input {
     pub dbc_id: DbcId,
-    pub blinded_amount: BlindedAmount,
+    pub amount: Amount,
     /// This is the signature of the `DerivedKey`
     /// corresponding to this `dbc_id`
     pub signature: Signature,
```

Comment:
It is the signature over the `input dbc_ids`, `input amount` and the `Vec<Output>`. 
i.e, it signs the message that we get from `fn serialize(dbc_ids: &[DbcId], input_amounts: &[Amount], outputs: &[Output]) -> Vec<u8>`

I assume we might not need it anymore? Since the `SignedSpend` has the signature over everything? 
If so, this might allow us to get rid of the `InputInteremediate` struct.

Reply:
Yes this is weird, we create a signature ourselves and then check it ourselves without the Network or anybody else ever seeing it... An artefact from the time when this job was done by elders and not clients?
If we have a sig in SignedSpend then this can go indeed.

Reply:
Neat, thanks for confirming it! This will simplify a lot of things!

Reply:
> It is the signature over the input dbc_ids, input amount and the Vec<Output>.
Could you add that description in the comment?